src/ai/backend/test/testcases/testcases.py

```diff
@@ -0,0 +1,76 @@
+import enum
+from dataclasses import dataclass
+from typing import Mapping, Self
+
+from ai.backend.test.testcases.template import (
+    BasicTestTemplate,
+    NopTestCode,
+    TestTemplate,
+)
+
+
+class TestTag(enum.StrEnum):
+    # component based tags
+    MANAGER = "manager"
+    AGENT = "agent"
+
+    # Domain specific tags
+    VFOLDER = "vfolder"
+    IMAGE = "image"
+    SESSION = "session"
```

Comment:
How about spliting TestTag into `ComponentTag` and `DomainTag`?

Reply:
I think it's not such a hassle to add one component tag and at least one domain tag.
And even if tags are managed as a single type, I think it would still be desirable to attach one component tag and at least one domain tag.

Reply:
In actual usage, it’ll be used in a form like `run-test -t {tag}`, and since there won’t be many other use cases, I’m not sure if strict categorization is really necessary.